src/framework/mlt_frame.h

```diff
@@ -150,6 +150,8 @@ extern int mlt_image_format_size( mlt_image_format format, int width, int height
 extern const char * mlt_audio_format_name( mlt_audio_format format );
 extern int mlt_audio_format_size( mlt_audio_format format, int samples, int channels );
 extern void mlt_frame_write_ppm( mlt_frame frame );
+extern int mlt_image_format_planes( mlt_image_format format, int width, int height, void* data, unsigned char *planes[4], int strides[4]);
```

Comment:
your approach make misunderstanding about width and stride, moreover, simple mlt_image_format_planes would be replaced with N calls: mlt_image_get_plane_count, mlt_image_get_plane_pointer(0..N),  mlt_image_get_plane_stride(0..N). I suggest to use ffmpeg approach that was checked for years...